recipes/botan/all/conanfile.py

```diff
@@ -117,84 +128,149 @@ def config_options(self):
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+            if not minimum_version:
+                self.output.warning(f"{self.name} recipe lacks information about the {compiler_name} compiler support.")
 
-        elif compiler == 'gcc' and version >= '5' and compiler.libcxx != 'libstdc++11':
+        if self.settings.compiler == 'gcc' and compiler_version >= "5" and self.settings.compiler.libcxx != 'libstdc++11':
             raise ConanInvalidConfiguration(
                 'Using Botan with GCC >= 5 on Linux requires "compiler.libcxx=libstdc++11"')
 
-        elif compiler == 'clang' and compiler.libcxx not in ['libstdc++11', 'libc++']:
+        if self.settings.compiler == 'clang' and self.settings.compiler.libcxx not in ['libstdc++11', 'libc++']:
             raise ConanInvalidConfiguration(
                 'Using Botan with Clang on Linux requires either "compiler.libcxx=libstdc++11" ' \
                 'or "compiler.libcxx=libc++"')
 
         # Some older compilers cannot handle the amalgamated build anymore
         # See also https://github.com/randombit/botan/issues/2328
-        if tools.Version(self.version) >= '2.14.0' and self.options.amalgamation:
-            if (compiler == 'apple-clang' and version < '10') or \
-               (compiler == 'gcc' and version < '8') or \
-               (compiler == 'clang' and version < '7'):
+        if Version(self.version) >= '2.14.0' and self.options.amalgamation:
+            if (self.settings.compiler == 'apple-clang' and compiler_version < '10') or \
+               (self.settings.compiler == 'gcc' and compiler_version < '8') or \
+               (self.settings.compiler == 'clang' and compiler_version < '7'):
                 raise ConanInvalidConfiguration(
-                    'botan amalgamation is not supported for {}/{}'.format(compiler, version))
+                    f"botan amalgamation is not supported for {compiler}/{compiler_version}")
 
         if self.options.get_safe("single_amalgamation", False) and not self.options.amalgamation:
             raise ConanInvalidConfiguration("botan:single_amalgamation=True requires botan:amalgamation=True")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data['sources'][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    @property
+    def _cxxflags(self):
+        global_cxxflags = " ".join(self.conf.get("tools.build:cxxflags", default=[], check_type=list))
+        env_cxxflags = VirtualBuildEnv(self).vars().get("CXXFLAGS", default="")
+        cxxflags = f"{env_cxxflags} {global_cxxflags}".strip()
+        return cxxflags if len(cxxflags) > 0 else None
```

Comment:
```suggestion
        return cxxflags if len(cxxflags) > 1 else None
```

because cxxflags will contain at least an empty-space always.

Reply:
Good to know!